src/config/read.js

```diff
@@ -66,6 +66,8 @@ function readRawConfig(argv, root) {
   }
 
   if (typeof argv.config === 'object') {
+    argv.config.name = argv.config.name || root.replace(/[/\\]|\s/g, '-');
+    argv.config.rootDir = argv.config.rootDir || root;
```

Comment:
It doesn't look like `loadFromFile` is called when passing the config as an object.